sdk/core/azure-core-http-netty/src/test/java/com/azure/core/http/netty/implementation/MockProxyServer.java

```diff
@@ -0,0 +1,124 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.http.netty.implementation;
+
+import com.azure.core.util.AuthorizationChallengeHandler;
+import com.azure.core.util.CoreUtils;
+import io.netty.handler.codec.http.HttpHeaders;
+import reactor.core.publisher.Flux;
+import reactor.netty.ConnectionObserver;
+import reactor.netty.DisposableServer;
+import reactor.netty.http.client.HttpClient;
+import reactor.netty.http.server.HttpServer;
+import reactor.netty.http.server.HttpServerRequest;
+import reactor.netty.http.server.HttpServerResponse;
+
+import java.io.Closeable;
+import java.net.InetSocketAddress;
+import java.net.SocketAddress;
+import java.util.Collections;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+
+import static com.azure.core.util.AuthorizationChallengeHandler.PROXY_AUTHENTICATE;
+
+/**
+ * This class represents a mock proxy server for testing.
+ */
+public final class MockProxyServer implements Closeable {
```

Comment:
Can this be moved to azure-core-test for other http clients to use?

Reply:
I think this could be moved into azure-core-test for a more widespread usage, it will force a Netty/Reactor Netty dependency though (but I think that may be for the better?).

Reply:
I was expecting this type to abstract away the exact http library being used and not take a hard dependency on netty.

Reply:
I'll write a follow-up issue to extend this further